### ISSUE TITLE
`rclone test info --all` always leaves files behind

### ISSUE THREAD

**Issue by kapitainsky** [rclone/rclone#7209]
`rclone test info --all remote:path` command should run all tests and clean any temporarily created files. This is not the case.

Start with empty directory/bucket:
```
$ rclone ls iDrive:test1
$
```

Run test:
```
$ rclone test info --all iDrive:test1
2023/08/04 12:16:31 NOTICE: S3 bucket test1: Streaming uploads using chunk size 5Mi will have maximum file size of 48.828Gi
2023/08/04 12:17:06 NOTICE: S3 bucket test1 path test-base32768: 0 differences found
2023/08/04 12:17:06 NOTICE: S3 bucket test1 path test-base32768: 1028 matching files
// iDrive
stringNeedsEscaping = []rune{
	'/', '\x00'
}
maxFileLength = 255 // for 1 byte unicode characters
maxFileLength = 127 // for 2 byte unicode characters
maxFileLength = 85 // for 3 byte unicode characters
maxFileLength = 63 // for 4 byte unicode characters
canWriteUnnormalized = true
canReadUnnormalized   = true
canReadRenormalized   = false
canStream = true
base32768isOK = true // make sure maxFileLength for 2 byte unicode chars is the same as for 1 byte characters
```

Have a look at thrash left behind:
```
$ rclone ls iDrive:test1
       50 ␁-position-left-01
       50 ␂-position-left-02
       50 ␃-position-left-03
...
       50 position-middle-01-␁-
       50 position-middle-02-␂-
       50 position-middle-03-␃-
...
       50 position-right-0A-␊
       50 position-right-0B-␋
       50 position-right-0C-␌

```

392 files left behind. I tested with different remotes and results are always the same.

Tested with v1.63.1 and latest v1.64 beta.

**Comment by kapitainsky** [comment]
Don't you consider it a bug? it is not really enhancement:)

**Comment by ncw** [comment]
Yes, `rclone test` is a bit messy and this would be nice to fix!

**Comment by ncw** [comment]
@KDreynolds that would be great.

Probably easiest to create a new random directory to run the tests in then operations.Purge it afterwards - we can then be sure we aren't deleting existing files.

We should have a flag which doesn't delete the files as sometimes it is useful to look at the files in testing.

**Comment by KDreynolds** [comment]
Ok I took a first stab at this with PR #7696 
Please lmk if any more changes are needed.

**Comment by ncw** [comment]
Thank you @KDreynolds 

I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org/) in 15-30 minutes and released in v1.67
